src/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -52,6 +54,11 @@ public CommandResult Execute()
                     // Reset the Reporters to the new Console Out and Error.
                     Reporter.Reset();
 
+                    if (!string.IsNullOrEmpty(_workingDirectory))
+                    {
+                        Directory.SetCurrentDirectory(_workingDirectory);
```

Comment:
This is needed because the only information passed into `compile-csc` is:  a string[] args and the current working directory.  `compile-csc` currently has no knowledge of the project.json or the project context information at all.

In order for `compile-csc` to launch `csc` in the directory that contains the project.json file, it needs to know which directory the project.json file is in.  This was the way I decided to pass that info into `compile-csc`.  Is there a better way?